docs/docs/contributing.md

```diff
@@ -0,0 +1,73 @@
+---
+slug: /contributing/
+sidebar_position: 7
+---
+
+# Contributing
+
+Contributions to Task are very welcome, but we ask that you read this document before submitting a PR.
+
+## Before you start
+
+- **Check existing work** - Is there an existing PR? Are there issues discussing the feature/change you want to make? Please make sure you consider/address these discussions in your work.
+- **Backwards compatibility** - Will your change break existing Taskfiles? It is much more likely that your change will merged if it backwards compatible. Is there an approach you can take that maintains this compatibility? If not, consider opening an issue first so that API changes can be discussed before you invest you time into a PR.
+
+## 1. Setup
+
+- **Golang** - Task is written in [Go]. You will need to [install version 1.18+].
+- **Node.js** - [Node.js] is used to host Task's documentation server and is required if you want to run this server locally.
+- **Yarn** - [Yarn] is the Node.js package manager used by Task.
+
+## 2. Making changes
+
+- **Code style** - Try to maintain the existing code style where possible and ensure that code is formatted by `gofmt`.
+- **Documentation** - Ensure that you add/update any relevant documentation. See the [updating documentation](#updating-documentation) section below.
+- **Tests** - Ensure that you add/update any relevant tests and that all tests are passing before submitting the PR. See the [writing tests](#writing-tests) section below.
+
+### Running my changes
+
+To run Task with working changes, you can use `go run ./cmd/task`. To run a development build of task against a test Taskfile in `testdata`, you can use `go run ./cmd/task --dir ./testdata/<my_test_dir> <task_name>`.
+
+### Updating documentation
+
+Task uses [Docusaurus] to host a documentation server. This can be setup and run locally by using `task docs:setup` and `task docs:start` respectively (requires `nodejs` & `yarn`). All content is written in Markdown and is located in the `docs/docs` directory.
+
+When making a change, consider whether a change to the [Usage Guide](./usage.md) is necessary. This document contains descriptions and examples of how to use Task features. If you're adding a new feature, try to find an appropriate place to add a new section. If you're updating an existing feature, ensure that the documentation and any examples are up-to-date. Ensure that any examples follow the [Taskfile Styleguide](./styleguide.md).
+
+If you added a new command/flag, ensure that you add it to the [API Reference](./api_reference.md).
+
+### Writing tests
+
+All of Task's test are held in the `task_test.go` file in the project root. Any new tests should also be added here. Most of these tests also have a subdirectory in the `testdata` directory where any Taskfiles/data required to run the tests are stored.
```

Comment:
We have a couple of other `_test.go` files, even if most tests are on the mentioned one. Also, creating new test files is not prohibited.

```suggestion
Most of Task's test are held in the `task_test.go` file in the project root. New tests should mostly likely be added here. Most of these tests also have a subdirectory in the `testdata` directory where any Taskfiles/data required to run the tests are stored.
```